server/api/call.go

```diff
@@ -7,36 +7,57 @@ import (
 	"github.com/mattermost/mattermost-plugin-apps/server/utils/md"
 )
 
+type CallType string
+
+const (
+	// CallTypeSubmit (default) indicates the intent to take action.
+	CallTypeSubmit = CallType("")
+	// CallTypeForm retrieves the form definition for the current set of falues,
+	// and the context.
+	CallTypeForm = CallType("form")
+	// CallTypeCancel is used for for the (rare?) case of when the form with
+	// SubmitOnCancel set is dismissed by the user.
+	CallTypeCancel = CallType("cancel")
+)
+
 type Call struct {
 	URL        string            `json:"url,omitempty"`
+	Type       CallType          `json:"type,omitempty"`
 	Values     map[string]string `json:"values,omitempty"`
```

Comment:
Something to keep in mind with using `map[string]string`, we will have multiselect values which will probably need to be sent as `[]string`. What are the pros of going with `map[string]string` as the container for the values, versus `map[string]interface{}`?

Reply:
ack. A separate change when we need it?